internal/exec/util/util.go

```diff
@@ -92,6 +92,35 @@ func (u Util) JoinPath(path ...string) (string, error) {
 	return filepath.Join(u.DestDir, realpath, last), nil
 }
 
+// NotateMkdirAll creates directories relative to the u.DestDir root,
+// including any missing parents, and records the paths to any created
+// directories in the State for future use.
+func (u Util) NotateMkdirAll(path string, perm os.FileMode) error {
```

Comment:
Did you see `FindFirstMissingDirForFile`? Feels like there's overlap here between the two. I think we could have this function just call that function instead?

Reply:
We still need to notate every subdirectory we've created, because the files stage needs to chown/chmod each one individually.  So I don't think it's worth it.